packages/flutter_test/lib/src/binding.dart

```diff
@@ -736,6 +739,40 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
   @override
   int get microtaskCount => _currentFakeAsync.microtaskCount;
 
+  static Set<String> _allowedKeys;
+
+  void _mockFlutterAssets() {
+    if (!Platform.environment.containsKey('UNIT_TEST_ASSETS')) {
+      return;
+    }
+    final String assetFolderPath = Platform.environment['UNIT_TEST_ASSETS'];
+    _ensureInitialized(assetFolderPath);
+    BinaryMessages.setMockMessageHandler('flutter/assets', (ByteData message) {
```

Comment:
Why not use the `setMethodCallHandler` method, then it wouldn't be a breaking change as the mocks would still take precedence.

Reply:
If I understand correctly, settMockMessageHandler is setting _mockHandlers, which is used for outgoing message. setMethodCallHandler is setting _handlers, which is for in coming message. The assets request is outgoing message. I think setting the _handlers will not be able to intercept the request?

Reply:
Ahh good point, my bad!